packages/botonic-react/src/components/multichannel/whatsapp/markdown.ts

```diff
@@ -1,105 +1,66 @@
-      if (match.startsWith(MARKDOWN_BOLD_OPTION_1)) {
-        return replaceAllOccurrences(
-          match,
-          MARKDOWN_BOLD_OPTION_1,
-          MARKDOWN_WHATSAPP_BOLD
-        )
-      }
-
-      if (match.startsWith(MARKDOWN_BOLD_OPTION_2)) {
-        return replaceAllOccurrences(
-          match,
-          MARKDOWN_BOLD_OPTION_2,
-          MARKDOWN_WHATSAPP_BOLD
-        )
-      }
-
-      if (match.startsWith(MARKDOWN_ITALIC_OPTION_1)) {
-        return replaceAllOccurrences(
-          match,
-          MARKDOWN_ITALIC_OPTION_1,
-          MARKDOWN_WHATSAPP_ITALIC
-        )
-      }
-
-      return match
-    })
-
-    let textWhatsapp = textNormalized
-    for (let i = 0; i < matches.length; i++) {
-      textWhatsapp = replaceAllOccurrences(
-        textWhatsapp,
-        matches[i],
-        matchesResult[i]
-      )
-    }
+function replaceItalicAndBold(text: string) {
+  const normalizedText = normalizeMarkdown(text)
+  const boldAndItalicText = convertNormalizedToWhatsApp(normalizedText)
 
-    return textWhatsapp
-  }
-
-  return text
+  return boldAndItalicText
 }
 
-function normalizeMarkdown(text: string) {
-  text = replaceBoldItalic(text, MARKDOWN_BOLD_AND_ITALIC_OPTION1)
-  text = replaceBoldItalic(text, MARKDOWN_BOLD_AND_ITALIC_OPTION2)
-  text = replaceBoldItalic(text, MARKDOWN_BOLD_AND_ITALIC_OPTION3)
+function normalizeMarkdown(text: string): string {
+  // Normalize bold
+  text = text.replace(
+    REGEX_MARKDOWN_BOLD,
+    `${NORMALIZED_BOLD}$2${NORMALIZED_BOLD}`
+  )
+  // Normalize italic
+  text = text.replace(
+    REGEX_MARKDOWN_ITALIC,
+    `${NORMALIZED_ITALIC}$2${NORMALIZED_ITALIC}`
+  )
+
   return text
 }
 
-function replaceBoldItalic(text: string, regex: RegExp) {
-  return text.replace(
-    regex,
-    (
-      match: string,
-      markdownOpen: string,
-      textInsideMarkdown: string,
-      markdownClose: string
-    ) => {
-      if (match.startsWith(markdownOpen) && match.endsWith(markdownClose)) {
-        return `${MARKDOWN_NORMALIZED_BOLD_ITALIC_OPEN}${textInsideMarkdown}${MARKDOWN_NORMALIZED_BOLD_ITALIC_CLOSE}`
-      }
-      return match
-    }
+function convertNormalizedToWhatsApp(text: string): string {
+  // convert &%BOLD%&text&%BOLD%& to *text*
+  text = text.replace(
+    REGEX_NORMALIZED_BOLD,
+    `${WHATSAPP_BOLD}$1${WHATSAPP_BOLD}`
+  )
+  // convert &%ITALIC%&text&%ITALIC%& to _text_
+  text = text.replace(
+    REGEX_NORMALIZED_ITALIC,
+    `${WHATSAPP_ITALIC}$1${WHATSAPP_ITALIC}`
   )
-}
 
-function replaceAllOccurrences(
-  text: string,
-  searchValue: string,
-  replaceValue: string
-) {
-  return text.split(searchValue).join(replaceValue)
+  return text
 }
 
-const REGEX_MARKDOWN_LINK = /\[([^\]]+)\]\(([^)]+)\)/g
-
-function replaceMarkdownLinks(markdown: string) {
-  return markdown.replace(REGEX_MARKDOWN_LINK, '$1: $2')
+function replaceMarkdownLinks(text: string) {
+  // $1 = textUrl, $2 = linkUrl
+  return text.replace(REGEX_MARKDOWN_LINK, '$1: $2')
 }
```

Comment:
I have looked about captured groups in regexs and you should be able to do this (it can be also applied in the other PR)
```
const REGEX_MARKDOWN_LINK = /\[(?<textUrl>[^\]]+)\]\((?<linkUrl>[^)]+)\)/g;

function replaceMarkdownLinks(text: string) {
  return text.replace(REGEX_MARKDOWN_LINK, '$<textUrl>: $<linkUrl>');
}
```

Reply:
Cannot be used with the configuration we have for the CommonJS build